pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -39,42 +42,43 @@ func (*AgentClusterInstall) Dependencies() []asset.Asset {
 
 // Generate generates the AgentClusterInstall manifest.
 func (a *AgentClusterInstall) Generate(dependencies asset.Parents) error {
-	// installConfig := &agent.OptionalInstallConfig{}
-	// dependencies.Get(installConfig)
-
-	// agentClusterInstall := &hiveext.AgentClusterInstall{
-	// 	ObjectMeta: metav1.ObjectMeta{
-	// 		Name:      "agent-cluster-install",
-	// 		Namespace: installConfig.Config.Namespace,
-	// 	},
-	// 	Spec: hiveext.AgentClusterInstallSpec{
-	// 		ClusterDeploymentRef: corev1.LocalObjectReference{
-	// 			Name: installConfig.Config.ObjectMeta.Name,
-	// 		},
-	// 		SSHPublicKey: installConfig.Config.SSHKey,
-	// 		ProvisionRequirements: hiveext.ProvisionRequirements{
-	// 			ControlPlaneAgents: int(*installConfig.Config.ControlPlane.Replicas),
-	// 		},
-	// 	},
-	// }
-
-	// var numberOfWorkers int = 0
-	// for _, compute := range installConfig.Config.Compute {
-	// 	numberOfWorkers = numberOfWorkers + int(*compute.Replicas)
-	// }
-	// agentClusterInstall.Spec.ProvisionRequirements.WorkerAgents = numberOfWorkers
-	// a.Config = agentClusterInstall
-
-	// agentClusterInstallData, err := yaml.Marshal(agentClusterInstall)
-	// if err != nil {
-	// 	return errors.Wrap(err, "failed to marshal agent installer AgentClusterInstall")
-	// }
-
-	// a.File = &asset.File{
-	// 	Filename: agentClusterInstallFilename,
-	// 	Data:     agentClusterInstallData,
-	// }
+	installConfig := &agent.OptionalInstallConfig{}
+	dependencies.Get(installConfig)
+
+	if installConfig.Config != nil {
```

Comment:
How could happen this condition? If the dependency gets successfully resolved shouldn't be always valid?

Reply:
This is added for this test case https://github.com/pawanpinjarkar/installer/blob/convert-install-config-to-ztp-assets/pkg/asset/agent/manifests/agentclusterinstall_test.go#L31-L35

In absence of this check the test case fails with below error -
```
--- FAIL: TestAgentClusterInstall_Generate (2.53s)
    --- FAIL: TestAgentClusterInstall_Generate/missing_install_config (2.53s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2ffb8d3]

goroutine 68 [running]:
testing.tRunner.func1.2({0x33c9c20, 0x578af20})
	/usr/local/go/src/testing/testing.go:1209 +0x47e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1212 +0x350
panic({0x33c9c20, 0x578af20})
	/usr/local/go/src/runtime/panic.go:1047 +0x262
github.com/openshift/installer/pkg/asset/agent/manifests.(*AgentClusterInstall).Generate(0xc00009ba08, 0xc00009ba68)
	/home/pawanpinjarkar/go/src/github.com/openshift/installer/pkg/asset/agent/manifests/agentclusterinstall.go:58 +0x113
github.com/openshift/installer/pkg/asset/agent/manifests.TestAgentClusterInstall_Generate.func1(0xc000603ba0)
	/home/pawanpinjarkar/go/src/github.com/openshift/installer/pkg/asset/agent/manifests/agentclusterinstall_test.go:66 +0x138
testing.tRunner(0xc000603ba0, 0xc0006a1d00)
	/usr/local/go/src/testing/testing.go:1259 +0x1db
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x673
```
Essentially, when the `Config` is empty and trying to retrieve values out of it.

Reply:
It's OptionalInstallConfig, so if the user doesn't provide one this will be nil. (You can also check the `installConfig.Supplied` flag, but the effect is that same and in retrospect I probably didn't need to create that.)

Reply:
Right, I didn't notice it initially :). Can you please remember me why it was required? I mean, as per the asset infrastructure during the fetch first of all it tries to load the file from the the disk, if not it goes to the Generate(). I'd say the if end up in a asset Generate() then the install-config must be present

Reply:
(so now that we're moving to fully support install-config maybe such test doesn't make sense anymore for an asset?)

Reply:
I guess we still need to decide what to do if the user doesn't supply an install-config _or_ ZTP manifests and they run "agent create cluster-manifests". Maybe the answer is to just return an error. (The answer is definitely not to panic though :laughing:)
It would be nice to generate a default template that users can fill in, but that might be a bit challenging for now (how good is our validation that they actually filled it in before creating the image?) so maybe that is a story for another day. I just created [AGENT-301](https://issues.redhat.com/browse/AGENT-301 to keep track of that.